src/ts/components/subtitleoverlay.ts

```diff
@@ -86,10 +101,88 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
       }
     });
 
+    this.configureCea608Captions(player, uimanager);
     // Init
     subtitleClearHandler();
   }
 
+  configureCea608Captions(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
+    let ratio = 1;
+    let updateCEA608FontSize = () => {
+      let dummyText = 'aaaaaaaaaa';
+      let label = new SubtitleLabel({
+        // One letter label used to calculate the height wifth ratio of the font
+        // Works because we are using a monospace font for cea 608
+        // Using a longer string increases precision due to width being an integer
+        text: dummyText,
+      });
+      let domElement = label.getDomElement();
+      domElement.css({
+        'color': 'rgba(0, 0, 0, 0)',
+        'font-size': '30px',
+        'line-height': '30px',
+        'top': '0',
+        'left': '0',
+      });
+      this.addComponent(label);
+      this.updateComponents();
+      this.show();
+
+      let width = domElement.width() / dummyText.length;
+      let height = domElement.height();
+
+      this.removeComponent(label);
+      this.updateComponents();
+      if (!this.subtitleManager.hasCues) {
+        this.hide();
+      }
+
+      ratio = height / width;
+      this.cea608lineHeight = (new DOM(player.getFigure()).width()) * (SubtitleOverlay.CEA608_WIDTH / SubtitleOverlay.CEA608_NUM_COLUMN) * ratio * SubtitleOverlay.CEA608_FONT_SPACING;
+
+      if (this.isCEA608) {
+        for (let label of this.getComponents()) {
+          if (label instanceof SubtitleLabel) {
+            // Only element with left property are cea-608
+            let domElement = label.getDomElement();
+            domElement.css({
+              'font-size': `${SubtitleOverlay.CEA608_LINE_TO_FONT_SIZE * this.cea608lineHeight}px`,
+            });
+          }
+        }
+      }
+    };
+    player.addEventHandler(player.EVENT.ON_PLAYER_RESIZE, updateCEA608FontSize);
+
+    player.addEventHandler(player.EVENT.ON_CUE_ENTER, (event: SubtitleCueEvent) => {
+      if (event.position == null) {
+        return;
+      }
+
+      let labels = this.subtitleManager.getCues(event);
+
+      if (!this.isCEA608) {
```

Comment:
What do you mean, remove the `CLASS_CEA_608` from the component if one label isn't `CEA-608`? That would break other label not yet removed if they are positioned.